pages/index.js

```diff
@@ -160,17 +189,24 @@ export default function HomePage({ onSubmit }) {
       </StyledWeather>
       <StyledCardSection>
         {displayedActivities.length > 0 ? (
-          displayedActivities.map((activity) => (
-            <Link key={activity._id} href={`/${activity._id}`}>
-              <ActivityCard
-                name={activity.name}
-                date={activity.date}
-                time={activity.time}
-                joined={activity.joined}
-                category={activity.category}
-              />
-            </Link>
-          ))
+          displayedActivities
+            .filter(
+              (activity) =>
+                appUserFriendsList &&
+                appUserFriendsList.length > 0 &&
+                appUserFriendsList.find((user) => user === activity.authorId)
```

Comment:
We could simplify this a bit with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). 

```suggestion
                appUserFriendsList?.find((user) => user === activity.authorId)
```